src/App.tsx

```diff
@@ -1,156 +1,140 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import {
+  createTodo, deleteTodo, getTodo, getTodos, updateTodo,
+} from './api/todos';
+import { Todo } from './types/Todo';
 import { UserWarning } from './UserWarning';
+import { NewTodoForm } from './components/NewTodoForm';
+import { FormMain } from './components/FormMain';
+import { FormFooter } from './components/FormFooter';
+import { FormErrors } from './components/FormErrors';
 
-const USER_ID = 0;
+const USER_ID = 6146;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [todo, setTodo] = useState<Todo | null>(null);
+  const [error, setError] = useState(false);
+  const [errorMsg, setErrorMsg] = useState('');
+  const [filter, setFilter] = useState('All');
+  const [newTodoTitle, setNewTodoTitle] = useState('');
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then(setTodos)
+      .catch(() => {
+        setError(true);
+        setErrorMsg('Unable to load todos');
+
+        setTimeout(() => {
+          setError(false);
+        }, 3000);
+      });
+  }, []);
+
+  useEffect(() => {
+    if (todo) {
+      getTodo(todo.id).then(setTodo);
+    }
+  }, [todo]);
+
+  const addTodo = (todoData: Omit<Todo, 'id'>) => {
+    createTodo(todoData)
+      .then(newTodo => setTodos([...todos, newTodo]))
+      .catch(() => setErrorMsg('Unable to add todo'));
+  };
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    addTodo({
+      title: newTodoTitle,
+      completed: false,
+      userId: USER_ID,
+    });
+
+    setNewTodoTitle('');
+  };
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
+  const visibleTodos = todos.filter((td) => {
+    switch (filter) {
+      case 'active':
+        return !td.completed;
+      case 'completed':
+        return td.completed;
+      default:
+        return true;
+    }
+  });
+
+  const removeTodo = (todoId: number) => {
+    deleteTodo(todoId)
+      .then(() => {
+        setTodos(todos.filter(td => td.id !== todoId));
+      })
+      .catch(() => setErrorMsg('Unable to delete todo'));
+  };
+
+  const todoUpdate = (todoToUpdate: Todo) => {
+    updateTodo(todoToUpdate)
+      .then(() => {
+        setTodos(
+          todos.map(td => {
```

Comment:
whts td?

Reply:
Short for todo, since I have a state with name todo I used td for these. In fact I can't use todo I get error